labonneboite/web/contact_form/views.py

```diff
@@ -203,6 +203,8 @@ def update_coordinates_form(form):
                     site_name=redirect_params.get('site_name'),
                     email=redirect_params.get('email'),
                     home_url=redirect_params.get('home_url'),
+                    suggest_update_jobs=True,
+                    params=urlencode(request.args),
```

Comment:
Cet object `params` me turlupine, et ça m'embête un peu de le passer tel-quel au template, puis aux views `contact_form.update_jobs_form` et `contact_form.update_coordinates_form`.

D'ailleurs je m'aperçois que c'est le même problème aux lignes 143 et 263.

En fait, on se retrouve à trimbaler d'une page à une autre plein de paramètres, parmi lesquels il y a en a certainement plein dont on n'a pas besoin. Par exemple, si l'utilisateur s'amuse à ajouter des paramètres arbitraires dans l'url, ils vont se retrouver dans les différentes vues. Ce qui n'est pas forcément grave en soi, mais n'empêche ça me perturbe, sans que j'arrive très bien à mettre le doigt sur le problème.

En fait si, ce qui me perturbe c'est qu'on ne sait pas exactement ce qu'il y a dans ce gros blob `request.args`. C'est un cas où "explicit is better than implicit".

Je vois deux moyens pour résoudre le problème : 
1. Extraire manuellement les champs nécessaires dans un dict pour les passer au template : mais ça peut être assez verbeux, et surtout ça risque de faire de la duplication de code.
2. Créer un objet `form` à partir des arguments et transformer le lien vers `update_jobs_form` en un formulaire. Cette solution me paraît la plus logique, sachant que cette vue est censée elle-même prendre des formulaires en entrée.

Reply:
Je choisis l'option 1 !